/jds/bin/diff -uprN nautilus-2.26.3.old/src/file-manager/fm-desktop-icon-view.c nautilus-2.26.3/src/file-manager/fm-desktop-icon-view.c
--- nautilus-2.26.3.old/src/file-manager/fm-desktop-icon-view.c	2009-05-18 20:44:27.000000000 +0100
+++ nautilus-2.26.3/src/file-manager/fm-desktop-icon-view.c	2009-07-13 11:36:17.871809000 +0100
@@ -592,6 +592,25 @@ fm_desktop_icon_view_init (FMDesktopIcon
 }
 
 static void
+action_help_callback (GtkAction *action,
+                      gpointer data)
+{
+        GError *error = NULL;
+        g_assert (FM_DIRECTORY_VIEW (data));
+
+        gtk_show_uri (gtk_widget_get_screen (GTK_WIDGET (data)),
+		       "ghelp:user-guide", gtk_get_current_event_time (),
+		       &error);
+
+        if (error) {
+                /* SUN_BRANDING */
+                eel_show_error_dialog (_("There was an error displaying help."), error->message,
+                                       NULL);
+                g_error_free (error);
+        }
+}
+
+static void
 action_new_launcher_callback (GtkAction *action, gpointer data)
 {
 	char *desktop_directory;
@@ -623,6 +642,52 @@ action_change_background_callback (GtkAc
 }
 
 static void
+action_change_resolution_callback (GtkAction *action,
+                                   gpointer data)
+{
+        g_assert (FM_DIRECTORY_VIEW (data));
+
+        nautilus_launch_application_from_command (gtk_widget_get_screen (GTK_WIDGET (data)),
+                                                  /* SUN_BRANDING */
+                                                  _("Resolution"),
+                                                  "gnome-display-properties",
+                                                  NULL,
+                                                  FALSE);
+}
+
+static void
+action_run_application_callback (GtkAction *action,
+                                   gpointer data)
+{
+	Display *display;
+	Atom gnome_panel_atom, run_atom;
+	XClientMessageEvent event;
+
+	display = XOpenDisplay(NULL);
+	if (display == NULL)
+		return;
+
+	gnome_panel_atom = XInternAtom(display, "_GNOME_PANEL_ACTION", False);
+	run_atom = XInternAtom(display, "_GNOME_PANEL_ACTION_RUN_DIALOG", False);
+
+	if (gnome_panel_atom != None && run_atom != None) {
+		event.type = ClientMessage;
+		event.window = DefaultRootWindow(display);
+		event.message_type = gnome_panel_atom;
+		event.format = 32;
+		event.data.l[0] = run_atom;
+		event.data.l[1] = (Time)(time(NULL) * 1000);
+
+		XSendEvent(display, event.window, False, StructureNotifyMask,
+				(XEvent *)&event);
+
+	}
+	XCloseDisplay(display);
+
+	return;
+}
+
+static void
 action_empty_trash_conditional_callback (GtkAction *action,
 					 gpointer data)
 {
@@ -691,6 +756,15 @@ real_update_menus (FMDirectoryView *view
 
 static const GtkActionEntry desktop_view_entries[] = {
 	/* name, stock id */
+	{ "About Solaris Desktop", "gtk-help",
+	  /* label, accelerator */
+	  /* SUN_BRANDING */
+	  N_("_Help"), NULL,
+	  /* tooltip */
+	  /* SUN_BRANDING */
+	  N_("About Sun Java Desktop System"),
+	  G_CALLBACK (action_help_callback) },
+	/* name, stock id */
 	{ "New Launcher Desktop", NULL,
 	  /* label, accelerator */
 	  N_("Create L_auncher..."), NULL,
@@ -698,13 +772,32 @@ static const GtkActionEntry desktop_view
 	  N_("Create a new launcher"),
 	  G_CALLBACK (action_new_launcher_callback) },
 	/* name, stock id */
-	{ "Change Background", NULL,
+	{ "Run Application", NULL,
+	  /* label, accelerator */
+	  /* SUN_BRANDING */
+	  N_("Run Application..."), NULL,
+	  /* tooltip */
+	  /* SUN_BRANDING */
+	  N_("Run an application by typing a command or choosing from a list"),
+	  G_CALLBACK (action_run_application_callback) },
+	/* name, stock id */
+	{ "Desktop Appearance", NULL,
 	  /* label, accelerator */
-	  N_("Change Desktop _Background"), NULL,
+	  /* SUN_BRANDING */
+	  N_("Desktop A_ppearance..."), NULL,
 	  /* tooltip */
 	  N_("Show a window that lets you set your desktop background's pattern or color"),
 	  G_CALLBACK (action_change_background_callback) },
 	/* name, stock id */
+	{ "Screen Resolution", NULL,
+	  /* label, accelerator */
+	  /* SUN_BRANDING */
+	  N_("Screen _Resolution..."), NULL,
+	  /* tooltip */
+	  /* SUN_BRANDING */
+	  N_("Show a window that lets you set your desktop screen resolution"),
+	  G_CALLBACK (action_change_resolution_callback) },
+	/* name, stock id */
 	{ "Empty Trash Conditional", NULL,
 	  /* label, accelerator */
 	  N_("Empty Trash"), NULL,
/jds/bin/diff -uprN nautilus-2.26.3.old/src/file-manager/nautilus-desktop-icon-view-ui.xml nautilus-2.26.3/src/file-manager/nautilus-desktop-icon-view-ui.xml
--- nautilus-2.26.3.old/src/file-manager/nautilus-desktop-icon-view-ui.xml	2009-04-20 11:57:20.000000000 +0100
+++ nautilus-2.26.3/src/file-manager/nautilus-desktop-icon-view-ui.xml	2009-07-13 11:36:17.872197000 +0100
@@ -8,8 +8,11 @@
                 </placeholder>
         </placeholder>
         <placeholder name="After Zoom Items">
+                        <menuitem name="Run Application" action="Run Application"/>
+                        <separator/>
                 <placeholder name="Background Items">
-                        <menuitem name="Change Background" action="Change Background"/>
+                        <menuitem name="Desktop Appearance" action="Desktop Appearance"/>
+			<menuitem name="Screen Resolution" action="Screen Resolution"/>
                 </placeholder>
         </placeholder>
 </popup>
